geoips/plugins/modules/procflows/single_source.py

```diff
@@ -1777,7 +1777,8 @@ def call(fnames, command_line_args=None):
         "Reading metadata from dataset with reader '%s'...", reader_plugin.name
     )
     xobjs = reader_plugin(fnames, metadata_only=True, **reader_kwargs)
-    source_name = xobjs["METADATA"].source_name
+    source_name = get_source_name(xobjs)
```

Comment:
See above comment - top level source_name should still be set in the reader, this file should not require modification.

Reply:
I refactored to keep source_name (and other required attributes) top level as you suggested.

Reply:
This change was reverted to keep source name at the top level.

Reply:
Thanks!